src/components/Element.js

```diff
@@ -10,98 +10,124 @@ type Props = {
   onBlur: Function,
   onFocus: Function,
   onReady: Function,
+
+  elements?: mixed,
 };
 
 const noop = () => {};
 
-const Element = (type: string, hocOptions: {sourceType?: string} = {}) => class extends React.Component {
-  static propTypes = {
-    elementRef: PropTypes.func,
-    onChange: PropTypes.func,
-    onBlur: PropTypes.func,
-    onFocus: PropTypes.func,
-    onReady: PropTypes.func,
-  }
-  static defaultProps = {
-    elementRef: noop,
-    onChange: noop,
-    onBlur: noop,
-    onFocus: noop,
-    onReady: noop,
-  }
-
-  static contextTypes = {
-    elements: PropTypes.object.isRequired,
-    registerElement: PropTypes.func.isRequired,
-    unregisterElement: PropTypes.func.isRequired,
-  }
-
-  constructor(props: Props, context: ElementContext) {
-    super(props, context);
-
-    const options = this._extractOptions(this.props);
-    this._element = this.context.elements.create(type, options);
-    this._setupEventListeners();
-    this._options = options;
-  }
-
-  componentDidMount() {
-    this._element.mount(this._ref);
-    if (hocOptions.sourceType) {
-      this.context.registerElement(hocOptions.sourceType, this._element);
-    }
-  }
-  componentWillReceiveProps(nextProps: Props) {
-    const options = this._extractOptions(nextProps);
-    if (Object.keys(options).length !== 0 && !shallowEqual(options, this._options)) {
+const Element = (type: string, hocOptions: {sourceType?: string} = {}) =>
+  class extends React.Component {
+    static propTypes = {
+      elementRef: PropTypes.func,
+      onChange: PropTypes.func,
+      onBlur: PropTypes.func,
+      onFocus: PropTypes.func,
+      onReady: PropTypes.func,
+
+      // For folks loading Stripe.js asynchronously:
+      elements: PropTypes.object,
+    };
+    static defaultProps = {
+      elementRef: noop,
+      onChange: noop,
+      onBlur: noop,
+      onFocus: noop,
+      onReady: noop,
+    };
+
+    static contextTypes = {
+      elements: PropTypes.object,
+      registerElement: PropTypes.func,
+      unregisterElement: PropTypes.func,
+    };
+
+    constructor(props: Props, context: ElementContext | {||}) {
+      super(props, context);
+
+      const options = this._extractOptions(this.props);
+
+      this._elements = this.context && this.context.elements
+        ? this.context.elements
+        : this.props.elements;
+
+      if (!this._elements || typeof this._elements.create !== 'function') {
+        throw new Error(
+          'If you are not using the StripeProvider and Elements components, please pass an `elements` instance (returned by stripe.elements()) to your Element.' // eslint-disable-line
```

Comment:
Will most people see this message because they forgot to wrap their elements with the StripeProvider?
Or do only people who use the async loading get there?